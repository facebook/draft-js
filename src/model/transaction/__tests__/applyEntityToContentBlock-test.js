/**
 * Copyright (c) 2013-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @emails oncall+ui_infra
 */

'use strict';

jest.autoMockOff();

const CharacterMetadata = require('CharacterMetadata');
const ContentBlock = require('ContentBlock');
const applyEntityToContentBlock = require('applyEntityToContentBlock');

const {
  List,
  Repeat,
} = require('immutable');

describe('applyEntityToContentBlock', () => {
  const block = new ContentBlock({
    key: 'a',
    text: 'Hello',
    characterList: List(Repeat(CharacterMetadata.EMPTY, 5)),
  });

  function getEntities(block) {
    return block.getCharacterList().map(c => c.getEntity()).toJS();
  }

  it('must apply from the start', () => {
    const modified = applyEntityToContentBlock(block, 0, 2, 'x');
    expect(getEntities(modified)).toEqual(['x', 'x', null, null, null]);
  });

  it('must apply within', () => {
    const modified = applyEntityToContentBlock(block, 1, 4, 'x');
    expect(getEntities(modified)).toEqual([null, 'x', 'x', 'x', null]);
  });

  it('must apply at the end', () => {
    const modified = applyEntityToContentBlock(block, 3, 5, 'x');
    expect(getEntities(modified)).toEqual([null, null, null, 'x', 'x']);
  });

  it('must apply to the entire text', () => {
    const modified = applyEntityToContentBlock(block, 0, 5, 'x');
    expect(getEntities(modified)).toEqual(['x', 'x', 'x', 'x', 'x']);
  });
});
