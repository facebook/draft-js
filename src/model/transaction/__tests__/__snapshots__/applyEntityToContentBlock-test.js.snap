// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must apply at the end 1`] = `
Object {
  "characterList": Array [
    Object {
      "entity": Array [],
      "style": Array [],
    },
    Object {
      "entity": Array [],
      "style": Array [],
    },
    Object {
      "entity": Array [],
      "style": Array [],
    },
    Object {
      "entity": Array [
        "x",
      ],
      "style": Array [],
    },
    Object {
      "entity": Array [
        "x",
      ],
      "style": Array [],
    },
  ],
  "data": Object {},
  "depth": 0,
  "key": "a",
  "text": "Hello",
  "type": "unstyled",
}
`;

exports[`must apply from the start 1`] = `
Object {
  "characterList": Array [
    Object {
      "entity": Array [
        "x",
      ],
      "style": Array [],
    },
    Object {
      "entity": Array [
        "x",
      ],
      "style": Array [],
    },
    Object {
      "entity": Array [],
      "style": Array [],
    },
    Object {
      "entity": Array [],
      "style": Array [],
    },
    Object {
      "entity": Array [],
      "style": Array [],
    },
  ],
  "data": Object {},
  "depth": 0,
  "key": "a",
  "text": "Hello",
  "type": "unstyled",
}
`;

exports[`must apply to the entire text 1`] = `
Object {
  "characterList": Array [
    Object {
      "entity": Array [
        "x",
      ],
      "style": Array [],
    },
    Object {
      "entity": Array [
        "x",
      ],
      "style": Array [],
    },
    Object {
      "entity": Array [
        "x",
      ],
      "style": Array [],
    },
    Object {
      "entity": Array [
        "x",
      ],
      "style": Array [],
    },
    Object {
      "entity": Array [
        "x",
      ],
      "style": Array [],
    },
  ],
  "data": Object {},
  "depth": 0,
  "key": "a",
  "text": "Hello",
  "type": "unstyled",
}
`;

exports[`must apply within 1`] = `
Object {
  "characterList": Array [
    Object {
      "entity": Array [],
      "style": Array [],
    },
    Object {
      "entity": Array [
        "x",
      ],
      "style": Array [],
    },
    Object {
      "entity": Array [
        "x",
      ],
      "style": Array [],
    },
    Object {
      "entity": Array [
        "x",
      ],
      "style": Array [],
    },
    Object {
      "entity": Array [],
      "style": Array [],
    },
  ],
  "data": Object {},
  "depth": 0,
  "key": "a",
  "text": "Hello",
  "type": "unstyled",
}
`;
